plugins/node/opentelemetry-instrumentation-pg/test/pg-pool.test.ts

```diff
@@ -288,5 +295,150 @@ describe('pg-pool@2.x', () => {
         assert.strictEqual(resNoPromise, undefined, 'No promise is returned');
       });
     });
+
+    describe('when specifying a responseHook configuration', () => {
+      const dataAttributeName = 'pg_data';
+      const query = 'SELECT 0::text';
+      const events: TimedEvent[] = [];
+
+      describe('AND valid responseHook', () => {
+        beforeEach(async () => {
+          const config: PgInstrumentationConfig = {
+            enhancedDatabaseReporting: true,
+            responseHook: (
+              span: Span,
+              data: pg.QueryResult | pg.QueryArrayResult
+            ) =>
+              span.setAttribute(
+                dataAttributeName,
+                JSON.stringify({ rowCount: data.rowCount })
+              ),
+          };
+
+          create(config);
+        });
+
+        it('should attach response hook data to resulting spans for query with callback ', done => {
+          const pgPoolattributes = {
+            ...DEFAULT_PGPOOL_ATTRIBUTES,
+          };
+          const pgAttributes = {
+            ...DEFAULT_PG_ATTRIBUTES,
+            [SemanticAttributes.DB_STATEMENT]: query,
+            [dataAttributeName]: '{"rowCount":1}',
+          };
+          const parentSpan = provider
+            .getTracer('test-pg-pool')
+            .startSpan('test span');
+          context.with(setSpan(context.active(), parentSpan), () => {
+            const resNoPromise = pool.query(query, (err, result) => {
+              if (err) {
+                return done(err);
+              }
+              runCallbackTest(
+                parentSpan,
+                pgPoolattributes,
+                events,
+                unsetStatus,
+                2,
+                0
+              );
+              runCallbackTest(
+                parentSpan,
+                pgAttributes,
+                events,
+                unsetStatus,
+                2,
+                1
+              );
+              done();
+            });
+            assert.strictEqual(
+              resNoPromise,
+              undefined,
+              'No promise is returned'
+            );
+          });
+        });
+
+        it('should attach response hook data to resulting spans for query returning a Promise', async () => {
+          const pgPoolattributes = {
```

Comment:
pgPoolattributes and pgAttributes can be moved as consts under the AND valid responseHook describe